core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -457,11 +484,85 @@ private synchronized PrioritizedSplitRunner pollNextSplitWorker()
         return null;
     }
 
+    @VisibleForTesting
```

Comment:
Remove as we only use it in the test code (Per the `VisibleForTesting` doc: `Annotates a program element that exists, or is more widely visible than otherwise necessary, only for use in test code`).

Reply:
Hey, this is needed in unit test: testTaskExecutorInterruptRunawaySplit().

I can't make this method private, since currently the static inner class "StuckSplitInterrupter" is private and not accessible for the unit test(which is intentional, since StuckSplitInterrupter is an internal logic of taskExecutor).